### PR TITLE
add viewport.get_camera_2d()

### LINKED ISSUE #5411: No 2D equivalent of viewport.get_camera()
I am making a 2D game and I need to get the current camera.
There is `get_viewport().get_camera()` but it is for 3D.
Is it implemented somewhere else, or is it missing?

### PR DESCRIPTION
**Pull request by verdog**
Fixes #5411
First time contributor, still learning how things fit together, so I'd be happy to make any adjustments needed. :slightly_smiling_face: 

The `viewport.get_camera()` functionality already exists in 3d, so I just mimicked how 3d keeps track of things and added it to 2d. I added a `Camera2D` pointer in the viewport class and it gets updated it as 2d cameras are set as current/turned off.

Unlike 3d, in 2d, it is not required for a camera to be current. If this is the case, `get_camera_2d()` just returns null (just like what calling `get_camera()` in 2d does).

### CODE REVIEW

**Comment by nanaian** [comment]
Possibly worth renaming `viewport.get_camera()` to `viewport.get_camera_3d()` to match the renaming of all nodes to have a 2D or 3D suffix in 4.0, alongside merging this.

**Comment by Phoenix1747** [comment]
Any progress with this PR?

**Comment by verdog** [comment]
I assume that by now another rebase will be needed to merge. Will happily do so if a review passes!

**Review by Calinou — APPROVED** [review]
I think this can be rebased so it can be tested on latest `master` first. The feature seems desired enough and was approved by reduz in the original issue, so it's probably good to merge if the code implementation is OK.

**Comment by verdog** [comment]
All fixed up. Just had to updated a few new instances of get_camera and a mention of it in the documentation.

**Comment by Phoenix1747** [comment]
Ping

**Review by KoBeWi — CHANGES_REQUESTED** [review]
I checked the code and it looks fine, except one thing. `set_current(false)` should check if this camera is really current, otherwise it will remove any current camera.

You can test this by creating two cameras, setting first to current and calling `set_current(false)` on the second one. `get_camera_2d()` will return null after this operation, while it should return the first camera.

**Comment by verdog** [comment]
Good catch! Fixed that.

**Review by KoBeWi — APPROVED** [review]

**Comment by ekumlin** [comment]
What is the status of this PR? It's been idle for a few months now but seems to be approved.

**Comment by Calinou** [comment]
> What is the status of this PR? It's been idle for a few months now but seems to be approved.

This pull request needs to be rebased against the latest `master` branch before it can merged.

File: scene/main/viewport.cpp
Comment (verdog) [inline scene/main/viewport.cpp:3737]: This popped up after the latest rebase, which raised an eyebrow. With the exception of camera_3d/2d, I don't think I touched anything here. Which implies that it was there when I originally branched, removed, and then added back from my leftover branch? Is this still correct?

Comment (KoBeWi) [reply scene/main/viewport.cpp:3737]: Initialization of these variables was moved to the header file. You should remove it and initialize `camera_2d` in the header.

Comment (verdog) [reply scene/main/viewport.cpp]: :+1:

**Comment by timvisee** [comment]
Note that this is a breaking, because it renames `Viewport.get_camera()` to `Viewport.get_camera_3d()`. I'm not sure whether that is desired.

**Comment by akien-mga** [comment]
> Note that this is a breaking change, because it renames `Viewport.get_camera()` to `Viewport.get_camera_3d()`. I'm not sure whether that is desired.

Since this targets 4.0, this is a good change, as it matches the rename of `Camera` to `Camera3D` (along with many other 3D nodes).

**Comment by timvisee** [comment]
> (...)
> 
> Since this targets 4.0, this is a good change, as it matches the rename of `Camera` to `Camera3D` (along with many other 3D nodes).

New to Godot. Thanks for clearing that up!

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And sorry for the delay reviewing and merging this PR :)

**Comment by biblical-text** [comment]
Is there a way to get_camera_2d() pre Godot 4? I kind of need to use this now.

**Comment by Calinou** [comment]
> Is there a way to get_camera_2d() pre Godot 4? I kind of need to use this now. (I need to get the current camera position, so i can apply the current camera position to the "about to become active" camera position)

For this feature to be available in Godot 3.5, this pull request needs to be remade against the `3.x` branch

**Pull request merged**
